### PR TITLE
In the REPL show extended explanations only once

### LINKED ISSUE #10943: Repl feature warnings are always verbose
```
scala> List(1,2,3) last
                   ^
       warning: postfix operator last should be enabled
       by making the implicit value scala.language.postfixOps visible.
       This can be achieved by adding the import clause 'import scala.language.postfixOps'
       or by setting the compiler option -language:postfixOps.
       See the Scaladoc for value scala.language.postfixOps for a discussion
       why the feature should be explicitly enabled.
res1: Int = 3

scala> List(1,2,3) last
                   ^
       warning: postfix operator last should be enabled
       by making the implicit value scala.language.postfixOps visible.
       This can be achieved by adding the import clause 'import scala.language.postfixOps'
       or by setting the compiler option -language:postfixOps.
       See the Scaladoc for value scala.language.postfixOps for a discussion
       why the feature should be explicitly enabled.
res2: Int = 3

scala> { List(1,2,3) last ; List(1,2,3) last }
                     ^
       warning: postfix operator last should be enabled
       by making the implicit value scala.language.postfixOps visible.
       This can be achieved by adding the import clause 'import scala.language.postfixOps'
       or by setting the compiler option -language:postfixOps.
       See the Scaladoc for value scala.language.postfixOps for a discussion
       why the feature should be explicitly enabled.
                                        ^
       warning: postfix operator last should be enabled
       by making the implicit value scala.language.postfixOps visible.
res3: Int = 3
```

### PR DESCRIPTION
**Pull request by som-snytt**
Fixes scala/bug#10943

Per-run reporting adds an extended explanation only once per feature warning.

REPL wants to warn once per resident session.

This PR hooks `featureWarning` to avoid multiple explanations, and also avoids repeated "extended explanations" in the reporter.

### CODE REVIEW

**Review by dwijnand — APPROVED** [review]

**Comment by dwijnand** [comment]
(Couldn't tell if it did already and it shouldn't or the other way round, so I tweaked the description.)

**Comment by som-snytt** [comment]
Updated to intercept the feature warning directly, where the REPL global remembers the "feature names" it has already warned about. (PerRunReporting remembers Symbols.)

Simplified the reporter-side processing, and left it operational for non-feature messaging with an "extended explanation". I don't know if there are any such messages currently.

**Comment by dwijnand** [comment]
What's the difference between the symbol and the name?  Is the featureTrait symbol not the symbol for the feature and as unique as its name?

File: src/compiler/scala/tools/nsc/Reporting.scala [hunk 1]
```diff
@@ -199,6 +199,8 @@ trait Reporting extends internal.Reporting { self: ast.Positions with Compilatio
     }
 
     private[this] var reportedFeature = Set[Symbol]()
+    protected def featureReported(featureTrait: Symbol): Unit = reportedFeature += featureTrait
```
Comment (dwijnand) [inline src/compiler/scala/tools/nsc/Reporting.scala:202]: Reuse in `featureWarning`?

Comment (som-snytt) [reply src/compiler/scala/tools/nsc/Reporting.scala:202]: I was lazy.

**Comment by som-snytt** [comment]
I didn't at-ask Lukas whether symbol was more specific than name, but I thought it's safer holding onto a string for the life of the global.

**Review by lrytz — APPROVED** [review]

**Pull request merged**